pkg/transport/transport.go

```diff
@@ -30,15 +31,30 @@ type Client interface {
 
 // Transport interface structure
 type Transport struct {
-	tlsOpts   *tlsopts.Options
-	observers []Observer
+	tlsOpts        *tlsopts.Options
+	observers      []Observer
+	requestTimeout time.Duration
+}
+
+// InvokeTimeout enables timeouts for an overly long method call
+type InvokeTimeout struct {
+	Timeout time.Duration
+}
+
+// Intercept adds a context timeout to a method call
+func (it InvokeTimeout) Intercept(ctx context.Context, method string, req interface{}, reply interface{},
+	cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+	timedCtx, cancel := context.WithTimeout(ctx, it.Timeout)
+	defer cancel()
+	return invoker(timedCtx, method, req, reply, cc, opts...)
 }
 
 // NewClient returns a newly instantiated Transport Client
-func NewClient(tlsOpts *tlsopts.Options, obs ...Observer) Client {
+func NewClient(tlsOpts *tlsopts.Options, requestTimeout time.Duration, obs ...Observer) Client {
```

Comment:
let's make a separate func `NewClientWithTimeout`, and make `NewClient` use the default defined in the package. Rest of the code will be clearer.

Reply:
ok sounds good. and should the default be on the transport struct or a const?

Reply:
making the default a const will be probably clearer

Reply:
ok 👍